### ISSUE TITLE
Allow LAN and Direct IP play without login

### ISSUE THREAD

**Issue by deee167** [ZeroK-RTS/Zero-K#4173]
It would be great if we could play LAN and direct IP without requiring a login to the servers.

Thanks

**Comment by Sharkgamedev** [comment]
I agree that it would be great, but not sure if it would be worth it?

**Comment by deee167** [comment]
> 
> 
> I agree that it would be great, but not sure if it would be worth it?

I think it would be worth it because I often want to play with my friends and people in the same house.  This way an online connection wouldn't be required.  

Also if the internet goes down you can still play.  This might happen frequently depending on where you live.

Isn't this just a wrapper GUI thing?  SpringRTS likely already supports this.  Isn't this how multiplayer works now, the wrapper likely just passes all the IP and connection parameters to SpringRTS?

**Comment by Licho1** [comment]
Springrts does not care where is it played so you could just send IPs to clients and server. Just create start scripts, you could do it even manually in a notepad.
Mixing it into way things currently work is however out of scope - though as always, patches are welcome!

**Comment by deee167** [comment]
> 
> 
> Springrts does not care where is it played so you could just send IPs to clients and server. Just create start scripts, you could do it even manually in a notepad.
> Mixing it into way things currently work is however out of scope - though as always, patches are welcome!

If it's that simple, why won't you just add another menu option to the GUI?

Can you give me a working example on hosting and joining a direct ip game with AI and 2 players with a given map from command line?

In any case, I would like to ask that it's added to the GUI since you say it's so simple.  You already have a GUI, it just needs a few changes.

**Comment by Sharkgamedev** [comment]
> Springrts does not care where is it played so you could just send IPs to clients and server. Just create start scripts, you could do it even manually in a notepad.
> Mixing it into way things currently work is however out of scope - though as always, patches are welcome!

I too am confused as to what you mean by, "Mixing it into way things currently work is however out of scope"?

**Comment by Licho1** [comment]
It means it's not a change we plan to implement. We cannot keep every "idea ticket" open. You can try modifying chobby to achieve your goal. 
Complications:
 - script of host needs to know who will be in game and on what team before starting it. You can add people mid-game but only spectators/rejoiners right now
 - you need to ensure all players have identical game content (no missing map etc)
 - UDP for spring needs to be open or hole-punched

**Comment by deee167** [comment]
> 
> 
> It means it's not a change we plan to implement. We cannot keep every "idea ticket" open. You can try modifying chobby to achieve your goal. If your solution works it could become part of the default game.
> 
> Complications:
> 
>     * script of host needs to know who will be in game and on what team before starting it. You can add people mid-game but only spectators/rejoiners right now
> 
>     * you need to ensure all players have identical game content (no missing map etc)
> 
>     * UDP for spring needs to be open or hole-punched

You don't need to worry about having the same version between all players, that should simplify things.  We just need a GUI like the one that already existing for hosting a multiplayer game.  This is in the game now.  This just needs to be tweaked to allow hosting and joining via direct IP, no need for even a lan game view.  Direct IP would be good progress already.

**Comment by Arch-Shaman** [comment]
Hi,

For the record, this repo is for the in game stuff. Like when youre actually playing the game.
Chobby handles all the lobby stuff and would probably be where you should open tickets like this. Though it might also belong in infra which handles the back end and site. 

Some of your issues will likely be closed in order to move them to the right repository, just a heads up. It isnt us disagreeing with you but just putting them in the right spot. I hope this explanation helps you. Please keep the tickets coming!

Thanks and have a good day.

**Comment by deee167** [comment]
> 
> 
> Hi,
> 
> For the record, this repo is for the in game stuff. Like when youre actually playing the game.
> Chobby handles all the lobby stuff and would probably be where you should open tickets like this. Though it might also belong in infra which handles the back end and site.
> 
> Some of your issues will likely be closed in order to move them to the right repository, just a heads up. It isnt us disagreeing with you but just putting them in the right spot. I hope this explanation helps you. Please keep the tickets coming!
> 
> Thanks and have a good day.

Okay, sorry about that.  Can you please link the new issue in this one?

**Comment by GoogleFrog** [comment]
Technically chobby could do it in the hacky way that was suggested. Make a ticket there.

**Comment by deee167** [comment]
> 
> 
> Technically chobby could do it in the hacky way that was suggested. Make a ticket there.

Can you please move the ticket where it needs to be.  I don't know how to move it will all the history.

Thanks

**Comment by GoogleFrog** [comment]
I don't think the history is worth keeping and I think having a ticket about it open is fine. Mentioning this ticket will be enough. Check for a duplicate before creating one.

**Comment by deee167** [comment]
> 
> 
> I don't think the history is worth keeping and I think having a ticket about it open is fine. Mentioning this ticket will be enough. Check for a duplicate before creating one.

I don't know where to create one.  I already made one here as it's for Zero-K.  Can you just make one where you need it to be please.

**Comment by Sharkgamedev** [comment]
> > I don't think the history is worth keeping and I think having a ticket about it open is fine. Mentioning this ticket will be enough. Check for a duplicate before creating one.
> 
> I don't know where to create one. I already made one here as it's for Zero-K. Can you just make one where you need it to be please.

I got you!

**Comment by deee167** [comment]
> 
> 
> > > I don't think the history is worth keeping and I think having a ticket about it open is fine. Mentioning this ticket will be enough. Check for a duplicate before creating one.
> > 
> > 
> > I don't know where to create one. I already made one here as it's for Zero-K. Can you just make one where you need it to be please.
> 
> I got you! In this case, one did previously exist.

Thanks.  Can you reference it here so I can follow the new issue.

**Comment by Sharkgamedev** [comment]
> > > > I don't think the history is worth keeping and I think having a ticket about it open is fine. Mentioning this ticket will be enough. Check for a duplicate before creating one.
> > > 
> > > 
> > > I don't know where to create one. I already made one here as it's for Zero-K. Can you just make one where you need it to be please.
> > 
> > 
> > I got you! In this case, one did previously exist.
> 
> Thanks. Can you reference it here so I can follow the new issue.

Yep, I've already referenced it - look above my last comment.

**Comment by deee167** [comment]
> 
> 
> > > > > I don't think the history is worth keeping and I think having a ticket about it open is fine. Mentioning this ticket will be enough. Check for a duplicate before creating one.
> > > > 
> > > > 
> > > > I don't know where to create one. I already made one here as it's for Zero-K. Can you just make one where you need it to be please.
> > > 
> > > 
> > > I got you! In this case, one did previously exist.
> > 
> > 
> > Thanks. Can you reference it here so I can follow the new issue.
> 
> Yep, I've already referenced it - look above my last comment.

lol, sorry about that.  I missed it.
